game/scripts/vscripts/units/stopfightingyourself.lua

```diff
@@ -0,0 +1,194 @@
+require('libraries/garbagecollection')
+
+-- Entry Function
+function Spawn(entityKeyValues)
+  local disabled = false
+  if disabled then
+    return
+  end
+
+  print("Starting AI for " .. thisEntity:GetUnitName() .. " " .. thisEntity:GetEntityIndex())
+  thisEntity:SetContextThink('StopFightingYourselfThink', partial(Think, thisEntity), 1)
+
+  ABILITY_dupe_heroes = thisEntity:FindAbilityByName('boss_stopfightingyourself_dupe_heroes')
+
+  thisEntity.GC = GarbageCollection:Add(thisEntity.illusions, 10)
+end
+
+function Think(state, target)
+  -- NOTE: I'm thinking too long
+  -- BUG: Not attacking
+  -- BUG: illusions do not get deleted correctly
```

Comment:
Illusions like to stay around in memory and then get reused when another illusion is created. It's a dota thing and it can cause lag in overly illusion heavy games. I think we're probably fine, and we can always tweak the max illusions if it becomes an issue.